### PR TITLE
Main ()=>dev

### PR DESCRIPTION
**Pull request by VadimBezhkov**
Merge ()=>{};

### CODE REVIEW

File: Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Choise.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace ConsoleApp14
+{
+    // My menu
+    enum Operation : byte
+    {
+        Resize = 1,
+        Rename,
+        Exit
+    }
+
```
Comment (Audyman) [inline Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Choise.cs]: extra line - need to remove

File: Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/ManagerProgramm.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+            }
+        }
+
+        //Default constructor
+        public ManagerProgramm()
+        {
+
```
Comment (Audyman) [inline Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/ManagerProgramm.cs]: extra line - need to remove

File: Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/ManagerProgramm.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            foreach (string image in files)
+            {
+                if (token.IsCancellationRequested)
+                {
+                    Console.Clear();
+                    Console.WriteLine(new string('-', 119));
+                    Console.WriteLine("Task canceled");
+                    Console.WriteLine(new string('-', 119));
+                    ManagerProgramm asd = new ManagerProgramm();
```
Comment (Audyman) [inline Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/ManagerProgramm.cs:100]: add some separators

File: Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Program.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace ConsoleApp14
+{
+
```
Comment (Audyman) [inline Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Program.cs]: see above

File: Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Program.cs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    class Program
```
Comment (Audyman) [inline Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Program.cs]: access modifier is required

File: Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        
+        public void RenameImages(CancellationToken token)
+        {
+            if ((!Directory.Exists(oldFilePath)) || (!Directory.Exists(newFilePath)))
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("Oops.We can't find this directory.");
+                
+                //Thinking about to throw new DirectoryNotFoundException here.
+                Thread.Sleep(3000);
```
Comment (Audyman) [inline Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs:23]: Why you did it? Need to remove!

File: Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+            Console.Write("Enter new name for all files: ");
+            string newNameForAllFilesInFolder = Console.ReadLine();
+            Console.ResetColor();
+            
+            int index = 1;
+            try
+            {
+                foreach (var item in files)
+                {
+                    item.Rename($"{newNameForAllFilesInFolder}.{index}.jpg");
```
Comment (Audyman) [inline Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs:40]: what about other extensions of image files? (png, bmp etc.)

File: Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+                    index++;
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("It was an exception. " + ex.Message);
+                Thread.Sleep(3000);
```
Comment (Audyman) [inline Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs:47]: see above

File: Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+                Console.Clear();
+            }
+            finally
+            {
+                Console.ForegroundColor = ConsoleColor.DarkCyan;
+                Console.WriteLine("Your operation was complete. Press any button to continue.");
+                Console.ReadKey();
+                Console.ResetColor();
+                Console.Clear();
```
Comment (Audyman) [inline Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs:56]: Can be moved to a separate private method

File: Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+            TryParseMethod(ref width,ref height);
+            Console.Clear();
+
+            if ((!Directory.Exists(oldFilePath)) || (!Directory.Exists(newFilePath)))
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("Oops.We can't find this directory.");
+                
+                //Thinking about to throw new DirectoryNotFoundException here.
+                Thread.Sleep(3000);
```
Comment (Audyman) [inline Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs:74]: see above

File: Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs [hunk 11]
```diff
@@ -0,0 +1,144 @@
+using System.IO;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Threading;
+using System.Threading.Tasks;
+
```
Comment (Audyman) [inline Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs:7]: this class shouldn't contain logics related to Console class

File: Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Program.cs [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+            while (processRun)
+            {
+                showMenu?.Invoke();
+                char choice = Console.ReadKey().KeyChar;
+                Console.Clear();
+
+                switch (choice)
+                {
+                    case '1':
+
```
Comment (Audyman) [inline Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Program.cs:28]: extra blank lines

**Review by Audyman — CHANGES_REQUESTED** [review]

File: Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Program.cs (hunk 4 above)
Comment (VadimBezhkov) [reply Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Program.cs]: image

Comment (VadimBezhkov) [reply Bezhkov/Bulk-Thumbnail-Creator/ConsoleApp14/Program.cs]: ???

File: Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs (hunk 8 above)
Comment (VadimBezhkov) [reply Yakubovsky/BulkThumbnailCreator/BulkThumbnailCreator/Finder.cs:47]: That's not mine

**Pull request closed without merging**